qmi/instruments/wl_photonics/wltf_n.py

```diff
@@ -0,0 +1,334 @@
+        """
+        super().__init__(context, name)
+        self._transport = create_transport(transport)
+
+        # Instrument ranges for values. Will be updated in `open()`
+        self._wavelength_range = _WavelengthRange
+        self._steps_range = _StepsRange
+
+        # Current motor position step value. Initialize with 0, it will be updated in `open()`
+        self._motor_position = 0
+
+        # For convenience
+        self.encoding = "ascii"
+        self.response_terminator = self.RESPONSE_TERMINATOR.decode(self.encoding)
+
+    def _write(self, cmd: str) -> None:
+        """Write a command to the instrument and check if any errors or alerts were raised.
+
+        Parameters:
+            cmd: The input command to send.
+        """
+        self._check_is_open()
+        cmd = cmd + self.CMD_TERMINATOR
+        self._transport.write(cmd.encode(self.encoding))
+
+    def _read(self) -> str:
+        """Read data from the instrument buffer."""
+        self._check_is_open()
+        response = self._transport.read_until(self.RESPONSE_TERMINATOR, timeout=self.DEFAULT_RESPONSE_TIMEOUT)
+        return response.decode(self.encoding)
+
+    def _clear_ok(self, cmd: str) -> None:
+        """Clear response 'OK' from the instrument buffer. If the returned value was something else than "OK",
+        the clear method will raise an exception.
+
+        Parameters:
+            cmd: The command that was sent in.
+
+        Raises:
+            QMI_InstrumentException: If the instrument returned another response than "OK".
+
+        """
+        response = self._read()
+        # Check that we got 'OK' as second item in response. The first is the echo of the command sent.
+        if response.split(self.response_terminator)[1] == "":
+            return
+
+        # Else try to check what went wrong
+        raise QMI_InstrumentException(f"Command {cmd} returned {response} instead of OK")
+
+    def _ask_int(self, cmd: str) -> int:
+        """Send a query and return an integer number response."""
+        self._write(cmd)
+        full_response = self._read().split("\n")  # First item should be the response, the second "OK".
+        try:
+            return int(search(r"[-\d.]+", full_response[0])[0])  # type: ignore
+        except ValueError as exc:
+            raise QMI_InstrumentException(f"Unexpected response to command {cmd!r}: {full_response!r}") from exc
+        except TypeError as exc:
+            raise QMI_InstrumentException(f"Unexpected response to command {cmd!r}: {full_response!r}") from exc
+
+    def _ask_float(self, cmd: str) -> float:
+        """Send a query and return a floating point response."""
+        self._write(cmd)
+        full_response = self._read().split("\n")  # First item should be the response, the second "OK".
+        if "Unknown" in full_response[0]:
+            # The wavelength has not been set or calibrated yet
```

Comment:
should we log a warning here?

Reply:
Yes, let's make use of the `_logger` here.